packages/compass-shell/src/components/compass-shell/tab-compass-shell.tsx

```diff
@@ -60,62 +57,48 @@ type CompassShellProps = {
   initialInput?: string;
 };
 
-function useInitialEval(initialEvaluate?: string | string[]) {
+function useInitialEval(
+  initialEvaluate: string | string[] | undefined,
+  isRender: boolean
+) {
   const [initialEvalApplied, setInitialEvalApplied] = useTabState(
     'initialEvalApplied',
     false
   );
   useEffect(() => {
-    setInitialEvalApplied(true);
-  }, [setInitialEvalApplied]);
+    // as soon as we render the first time, set it to true
+    if (isRender && !initialEvalApplied) {
+      setInitialEvalApplied(true);
+    }
+  }, [initialEvalApplied, setInitialEvalApplied, isRender]);
   return initialEvalApplied ? undefined : initialEvaluate;
 }
 
-const Shell = React.forwardRef<ShellType, ShellProps>(function Shell(
-  { initialEvaluate: _initialEvaluate, ...props },
-  ref
-) {
-  const shellRef = useRef<ShellType | null>(null);
-  const initialEvaluate = useInitialEval(_initialEvaluate);
-  const mergeRef = useCallback(
-    (shell: ShellType | null) => {
-      shellRef.current = shell;
-      if (typeof ref === 'function') {
-        ref(shell);
-      } else if (ref) {
-        ref.current = shell;
-      }
-    },
-    [ref]
-  );
-  useEffect(() => {
-    return rafraf(() => {
-      shellRef.current?.focusEditor();
-    });
-  }, []);
-  return (
-    <_Shell
-      ref={mergeRef}
-      initialEvaluate={initialEvaluate}
-      {...props}
-    ></_Shell>
-  );
-});
-
-const CompassShell: React.FC<CompassShellProps> = ({
+export const CompassShell: React.FC<CompassShellProps> = ({
   runtime,
   initialHistory,
   onHistoryChange,
-  initialEvaluate,
+  initialEvaluate: _initialEvaluate,
   initialInput,
 }) => {
   const enableShell = usePreference('enableShell');
-  const shellRef: ShellRef = useRef(null);
+  const canRenderShell = !!(enableShell && initialHistory && runtime);
+
+  // initialEvaluate will only be set on the first render
```

Comment:
Nit: now that it's not tied to the React lifecycle of the browser-repl component, I'd maybe clarify that this is specifically a first render of the broser-repl Shell, not this component first render